virtual_ecosystem/models/animals/animal_model.py

```diff
@@ -178,7 +178,13 @@ def spinup(self) -> None:
     def update(self, time_index: int, **kwargs: Any) -> None:
         """Function to step the animal model through time.
 
-        Currently this is a toy implementation.
+        Temporary solution.
```

Comment:
Depending on how much interdependence there is between the different events, we might want to use [coroutines ](https://docs.python.org/3.11/library/asyncio-task.html)here.

Reply:
Oh no... I mean that sounds appropriate but `asyncio` is scary! 😱

Reply:
It would be interesting to test it out and see how it changes the outcomes.